go/contacts/lookup.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package contacts
+
+import (
+	"time"
+
+	"github.com/keybase/client/go/emails"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/phonenumbers"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/client/go/uidmap"
+)
+
+type BulkLookupContactsProvider struct {
+}
+
+func (c *BulkLookupContactsProvider) LookupPhoneNumbers(mctx libkb.MetaContext, numbers []keybase1.RawPhoneNumber,
+	userRegion keybase1.RegionCode) (res []ContactLookupResult, err error) {
```

Comment:
@heronhaye there already is a provider that uses real API. But the cache layer is missing.